cmake/simd.cmake

```diff
@@ -57,43 +57,46 @@ int main()
     return 0;
 }" SSE3_FOUND)
 
-# Check AVX
-set(CMAKE_REQUIRED_FLAGS ${AVX_FLAG})
-set(AVX_FOUND_EXITCODE 1 CACHE STRING "Result from TRY_RUN" FORCE)
-CHECK_CXX_SOURCE_RUNS("
-#include <immintrin.h>
-int main()
-{
-    __m256 a = _mm256_set_ps (-1.0f, 2.0f, -3.0f, 4.0f, -1.0f, 2.0f, -3.0f, 4.0f);
-    __m256 b = _mm256_set_ps (1.0f, 2.0f, 3.0f, 4.0f, 1.0f, 2.0f, 3.0f, 4.0f);
-    __m256 result = _mm256_add_ps (a, b);
-    return 0;
-}" AVX_FOUND)
+# disable AVX by default on windows
+if(NOT WIN32)
```

Comment:
why disable avx?  windows has full of avx support, this check also works in windows platform.

Reply:
https://github.com/PaddlePaddle/Paddle/pull/14046/files#diff-971854df240cbe50b0f1ca7d316691d6R20

Reply:
some avx code still depends on legacy code, and better make a better plan to remove all dependency on legacy part instead of make some platform pre-definition macro to make it works. 

avx_functions.cc depends on paddle/legacy/cuda/src/avx_mathfun.h